src/main/java/operations/FilterOperation.java

```diff
@@ -116,28 +116,36 @@ private boolean constraintFunction(Value recordValue,
     }
   }
 
-  /**
-   * 
-   * execute the operation. before the operation can be executed the setConstraint() function must
-   * be called. this function fails if this.comparator is not set or set to ND
-   * 
-   * @return <b>true</b> iff the operation executed correctly<br>
-   *         <b>false</b> iff the operation did not execute correctly<br>
-   * 
-   * @see {@link setConstraint()}
-   * 
-   */
-  @Override
-  public boolean execute() {
-    for (Record record : this.inputData) {
-      if (record.containsKey(this.columnName)
-          && this.constraintFunction(record.get(this.columnName), this.constraintType,
-              this.constraintValue)) {
-        this.resultData.add(record);
-      }
-    }
-    return true;
-  }
+	/**
+	 * 
+	 * execute the operation. before the operation can be executed the
+	 * setConstraint() function must be called. this function fails if
+	 * this.comparator is not set or set to ND
+	 * 
+	 * @return <b>true</b> iff the operation executed correctly<br>
+	 *         <b>false</b> iff the operation did not execute correctly<br>
+	 * 
+	 * @see {@link setConstraint()}
+	 * 
+	 */
+	@Override
+	public boolean execute() {
+		if (!operationParametersSet) {
+		  return false;
+		} 
+	  for (Record record : this.inputData) {
+			if (record.containsKey(this.columnName)) {
+        if (this.constraintFunction(record.get(this.columnName), this.constraintType,
+            this.constraintValue)) {
+					this.resultData.add(record);
+				}
+			}
+			else {
```

Comment:
Add the record to the result dataset even when it does not meet the filter criteria, this is incorrect behavior